### PR TITLE
CPU Mining: Inject nonce bytes at the end

### PR DESCRIPTION
**Pull request by fosskers**
In the place of the (soon to be) former feature flags.

See #954

### CODE REVIEW

**Comment by fosskers** [comment]
`chainweb-miner` also relies on this code, so I'm going to test that now.

**Comment by fosskers** [comment]
This appears to work. I'm doing CPU mining to devnet, and I see my balance increasing. So, my byte offset value must be correct.

File: src/Chainweb/Miner/Core.hs [hunk 1]
```diff
@@ -169,8 +169,10 @@ mine _ orig@(Nonce o) (TargetBytes tbytes) (HeaderBytes hbytes) = do
 -- hashed `BlockHeader`. If that layout changes, this functions need to be
 -- updated. The assumption allows us to iterate on new nonces quickly.
 --
+-- Recall: `Nonce` contains a `Word64`, and is thus 8 bytes long.
+--
 injectNonce :: Nonce -> Ptr Word8 -> IO ()
-injectNonce (Nonce n) buf = poke (castPtr buf) n
+injectNonce (Nonce n) buf = pokeByteOff (castPtr buf) 278 n
```
Comment (fosskers) [inline src/Chainweb/Miner/Core.hs]: Seems to be the correct magic number.

Comment (larskuhtz) [reply src/Chainweb/Miner/Core.hs]: Maybe adding a comment with a reference to https://github.com/kadena-io/chainweb-node/wiki/Block-Header-Binary-Encoding ?

**Pull request merged**